colorcoder/ColorPairImpl.java

```diff
@@ -0,0 +1,29 @@
+package colorcoder;
+
+public class ColorPairImpl {
+
+	private ColorPairImpl() {
+		
+	}
+	
+	public static ColorPair getColorFromPairNumber(int pairNumber) {
+		int zeroBasedPairNumber = pairNumber - 1;
+		MajorColor majorColor = MajorColor.fromIndex(zeroBasedPairNumber / Main.NUMBEROFMINORCOLORS);
+		MinorColor minorColor = MinorColor.fromIndex(zeroBasedPairNumber % Main.NUMBEROFMINORCOLORS);
+		return new ColorPair(majorColor, minorColor);
+	}
+
+	public static int getPairNumberFromColor(MajorColor major, MinorColor minor) {
+		return major.getIndex() * Main.NUMBEROFMINORCOLORS + minor.getIndex() + 1;
+	}
+	
+	static void printManual() {

```

Comment:
The filename doesn't suggest that it will hold the printing functionality... Think of a more intuitive file name